src/lib/client/AuthorityClient.spec.ts

```diff
@@ -0,0 +1,234 @@
+import { jest } from '@jest/globals';
+
+import { MockCommand } from '../../testUtils/MockCommand.js';
+import { mockSpy } from '../../testUtils/jest.js';
+import type { PostRequest } from '../utils/requests.js';
+import { JsonValue } from '../utils/serialisation/JsonValue.js';
+
+import { AuthorityClient } from './AuthorityClient.js';
+import type { AuthorizationHeader } from './AuthorizationHeader.js';
+import { ServerError } from './ServerError.js';
+import { ClientError } from './ClientError.js';
+
+const mockFetch = mockSpy(jest.spyOn(global, 'fetch'));
+beforeEach(() => {
+  mockFetch.mockResolvedValue(new Response('{}'));
+});
+
+describe('AuthorityClient', () => {
+  const baseUrl = 'https://api.veraid-authority.example';
+  const authHeader: AuthorizationHeader = {
+    scheme: 'Bearer',
+    parameters: 'test',
+  };
+
+  const request: PostRequest = {
+    method: 'POST',
+    path: '/orgs',
+    body: new JsonValue({ foo: 'bar' }),
+  };
+
+  describe('send', () => {
+    describe('Request', () => {
+      test('Method should be honoured', async () => {
+        const client = new AuthorityClient(baseUrl, authHeader);
+
+        await client.send(new MockCommand(request));
+
+        expect(mockFetch).toHaveBeenCalledWith(
+          expect.anything(),
+          expect.objectContaining({ method: request.method }),
+        );
+      });
+
+      test('Path should be prefixed with API base URL', async () => {
+        const client = new AuthorityClient(baseUrl, authHeader);
+
+        await client.send(new MockCommand(request));
+
+        expect(mockFetch).toHaveBeenCalledWith(`${baseUrl}${request.path}`, expect.anything());
+      });
+
+      test('Any trailing slashes should be removed from the base URL', async () => {
+        const client = new AuthorityClient(`${baseUrl}/`, authHeader);
```

Comment:
Maybe test with multiple slashes: ```new AuthorityClient(`${baseUrl}//`, authHeader);```

Reply:
That's effectively what we're doing, since `request.path` begins with a `/`.

Reply:
In the functionality, we are removing the trailing slashes from the baseUrl not the path. Here we are just testing just one slash, so the regex is not really tested well since this also works in this case /\/$/u

Reply:
I see! Yeah, agreed. Will add a test.

Reply:
Maybe this is better there?
```     
 test.each([['slash', '/'], ['slashes', '//']])
 ('Any trailing %s should be removed from the base URL', async (_type: string, slashes: string) => {
        const client = new AuthorityClient(`${baseUrl}${slashes}`, authHeader);

        await client.send(new MockCommand(request));

        expect(mockFetch).toHaveBeenCalledWith(`${baseUrl}${request.path}`, expect.anything());
      });
      ```

Reply:
You can make this change or not. I'll approve the PR and you can resolve this conversation in case you are not making any change

Reply:
Done.